core/generalise.ml

```diff
@@ -174,8 +174,14 @@ let mono_type_args : type_arg -> unit =
 (** generalise:
     Universally quantify any free type variables in the expression.
 *)
-let generalise : gen_kind -> ?unwrap:bool -> environment -> datatype -> ((Quantifier.t list * type_arg list) * datatype) =
-  fun kind ?(unwrap=true) env t ->
+let generalise :
+      gen_kind ->
+      rigidify_vars:bool ->
```

Comment:
Not sure what the point of labelling this parameter is. If you insist on it being labelled, then I suggest moving it to the front of the parameter list. You can move `unwrap` too whilst you are at it.

Reply:
I usually prefer naming boolean parameters.
The reason for the unconventional ordering of parameters here was that the definition of `generalise_rigid`, `generalise`, ... further down in the file need this order for currying. I solved this now by making the definitions in lines ~235 slightly more verbose.